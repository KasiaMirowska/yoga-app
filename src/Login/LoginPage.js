import React from 'react';
import LoginForm from './LoginForm';

export default class LoginPage extends React.Component {
  
    handleLoginSuccess = () => {
      const { location, history } = this.props;
      const destination = (location.state || {}).from || '/';
      history.push(destination);
    }
  
    render() {
      return (
       <div className='current-flow'>
          <h2 className='title' >LOGIN</h2>
          <LoginForm
            onLoginSuccess={this.handleLoginSuccess}
          />
       </div>
      )
    };
  };