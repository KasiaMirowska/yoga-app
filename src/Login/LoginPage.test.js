import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import LoginPage from './LoginPage';
import {BrowserRouter as BR} from 'react-router-dom';

describe('LoginPage component', () => {
    // const props = {match: {params: {path: '/login'}}}
    
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><LoginPage /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const loginPage = renderer.create(<BR><LoginPage /></BR>);
        expect(loginPage.toJSON()).toMatchSnapshot();
    })

})