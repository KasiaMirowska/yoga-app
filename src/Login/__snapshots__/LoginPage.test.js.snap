// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renders UI as expected 1`] = `
<div>
  <h2>
    Login
  </h2>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="userName"
      >
        Username
      </label>
      <input
        id="userName"
        type="text"
      />
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        type="text"
      />
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
